### PR TITLE
function for computing wrapper PI + test

### PR DESCRIPTION
**Pull request by olesHolem**
# What ❔

Function for computing wrapper PI + test for it

### CODE REVIEW

File: src/proof_wrapper_utils.rs [hunk 1]
```diff
@@ -601,3 +604,27 @@ fn get_trusted_setup() -> Crs<Bn256, CrsForMonomialForm> {
...
+    for chunk in stark_pi.iter().rev() {
+        let mut chunk_fr =
+            Fr::from_repr(<Fr as PrimeField>::Repr::from(chunk.as_u64_reduced())).unwrap();
+        chunk_fr.mul_assign(&coeff);
+        result.add_assign(&chunk_fr);
+        coeff.mul_assign(&shift);
+    }
+
+    result
+}
```
Comment (mm-zk) [inline src/proof_wrapper_utils.rs]: can you add the unittest just for this function ?

Comment (olesHolem) [reply src/proof_wrapper_utils.rs]: Sure

File: src/proof_wrapper_utils/compression.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    let circuit_type = circuit.numeric_circuit_type();
+
+    test_compression_circuit(circuit.clone());
+    println!("Circuit is satisfied");
+
+    let proof_config = circuit.proof_config_for_compression_step();
+
+    let (setup_base, setup, vk, setup_tree, vars_hint, wits_hint, finalization_hint) =
+        create_compression_layer_setup_data(
+            circuit.clone(),
```
Comment (shamatar) [inline src/proof_wrapper_utils/compression.rs]: delete witness here

File: src/proof_wrapper_utils/compression.rs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            &worker,
+            proof_config.fri_lde_factor,
+            proof_config.merkle_tree_cap_size,
+        );
+
+    // prove
+    println!("Proving!");
+
+    let proof = prove_compression_layer_circuit::<NoPow>(
+        circuit.clone(),
```
Comment (shamatar) [inline src/proof_wrapper_utils/compression.rs]: You do not need clone

**Review by shamatar — DISMISSED** [review]

File: src/proof_wrapper_utils/compression.rs (hunk 3 above)
Comment (olesHolem) [reply src/proof_wrapper_utils/compression.rs]: `circuit` is still used in `verify_compression_layer_proof`, but I can remove this check

Comment (shamatar) [reply src/proof_wrapper_utils/compression.rs]: Ok, never mind then

**Review by shamatar — APPROVED** [review]

**Pull request merged**